resultsdbupdater/utils.py

```diff
@@ -426,6 +426,23 @@ def handle_ci_umb(msg):
             'system_provider': msg.system('provider', default=None),
         }
 
+    elif item_type == 'brew-build-group':
+        item = msg.get('artifact', 'id')
```

Comment:
Is there a way we can validate that the ID is in fact derived from the builds?

Reply:
This just needs to be unique enough so it can be used as subject ID in Greenwave.

@thrix Can you verify that using artifact.id from brew-build-group messages as subject identifier in Greenwave would work?

Reply:
uf, the current definition sucks:
https://pagure.io/fedora-ci/messages/blob/master/f/schemas/rpm-build-group.yaml#_8

We need to udpate this to something reasonable. Possibilities are "brew event ID" or something like @lcarva is suggesting

Reply:
afaik from Koji guys, brew event IDs should provide a unique id fo the sidetag in time .... this is quite nice as it can directly point to the sidetag in Brew

Reply:
Either is fine with me. Is there a way to easily get the event ID?

Reply:
From ResultsDB-Updater perspective, is it OK to use the ID for new result from brew-build-group message's `id` field (under `artifact`)? It would be ideal if Updater doesn't have to construct new ID from other fields.

Reply:
I think that's fine. We can revisit this if it becomes problematic.

Reply:
Ok, seems I still need to find if we can somehow get the event id

Reply:
so listing a sidetag for an event, should be easy as passing the event ID:

```
$ koji list-tagged --help
Usage: koji list-tagged [options] tag [package]
(Specify the --help global option for a list of other help options)

Options:
  -h, --help      show this help message and exit
  --arch=ARCH     List rpms for this arch
  --rpms          Show rpms instead of builds
  --inherit       Follow inheritance
  --latest        Only show the latest builds/rpms
  --latest-n=N    Only show the latest N builds/rpms
  --quiet         Do not print the header information
  --paths         Show the file paths
  --sigs          Show signatures
  --type=TYPE     Show builds of the given type only.  Currently supported
                  types: maven, win, image
  --event=EVENT#  query at event
  --ts=TIMESTAMP  query at timestamp
  --repo=REPO#    query at event for a repo
```

Reply:
@lcarva @hluk we need to find somebody from brew/koji mainainters how to get the event ID,
I can just guess it is via:

https://gist.github.com/amitsaha/4123768#file-koji-api-L1168

Reply:
It would be better if the event ID is already in the message received by updater. To keep resultsdb-updater simple and more reliable it should NOT query brew/koji.

Reply:
I agree, it will be, we just need to find out how hard is to get it. It is true that for resultsdb updater this really does not matter what really the id is ;)

Reply:
so, I can find this out out of band of this PR :)

Reply:
Oh, I'm OK with merging this if there won't be any major changes in the schema affecting the current code in updater. :)

Reply:
I think there will be 2 changes:
1. change current artitfact.id for event id
2. add the human readable sidetag name to the message format (maybe it could be even optional)

Discussing now with @pingou if this is ok

Reply:
In Fedora, in https://apps.fedoraproject.org/datagrepper/id?id=2019-185dbb94-4f2c-44b5-a6cc-8a4b921688c8&is_raw=true&size=extra-large artifact.id already an unique identifier,
it's the update's ID + the hash of its associated builds, so if the builds in the update change, the hash changes and thus artifact.id changes, otherwise it's stable

Reply:
hey guys, sorry for late reply ...

so we discussed this with @ralphbean and having the id as the hash of associated builds is fine, event id from brew is dead end, no benefits using it. So let's stick with that as the default. 

@pypingou we do not like much the concatenation with Ralph :) could we put the bodhi id as a separate field `artifact.label` maybe? and leave the artifact id as a hash of associated builds (what is a generic concept we use in the specification)

Reply:
Well the idea is that bodhi will need to query greenwave for a specific identifier, the concatenation gives 1 field instead of two if we split them (+ I guess changing this now would break backward compatibility on the message schema :()

Reply:
so... for resultsdb-updater, do we want to just take the ID as provided in the message, or do we want to automatically generate it based on the given builds?

IMO, resultsdb-updater should do some sort of verification to ensure the ID corresponds to the builds in the group.

Reply:
We will discuss this today on UMB consortium meeting and reply back

Reply:
> so... for resultsdb-updater, do we want to just take the ID as provided in the message, or do we want to automatically generate it based on the given builds?
> IMO, resultsdb-updater should do some sort of verification to ensure the ID corresponds to the builds in the group

I do not care much, but verification would be nice

Reply:
so we did not get to it :( but the overall agreement is to use sha256 of the sum of the subartifacts

Reply:
There will be a pagure.io/fedora-ci/message PR for this, let's continue the discussion there. If we would decide that resultsdb-updater will do a hash verification, we can add that later ...